In [1]:
!pip install accelerate protobuf sentencepiece torch git+https://github.com/huggingface/transformers -q
!pip install huggingface_hub accelerate protobuf sentencepiece torch -q
!pip install git+https://github.com/huggingface/transformers huggingface_hub -q

  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done


In [2]:
import sys

In [3]:
import torch
from transformers import AutoModelForCausalLM, AutoTokenizer, TextIteratorStreamer, pipeline

In [4]:
# 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
from huggingface_hub import login
login()

# Loading Fine-tuned Model



In [5]:
if torch.cuda.is_available():
    model_id = "asmaa-ali/llama-2-7b-AsmaaChemProt"
    model = AutoModelForCausalLM.from_pretrained(model_id, torch_dtype=torch.float16, device_map="auto")
    model.cuda()
    tokenizer = AutoTokenizer.from_pretrained(model_id)
    tokenizer.use_default_system_prompt = False

/usr/local/lib/python3.10/dist-packages/huggingface_hub/utils/_token.py:89: UserWarning: 
The secret `HF_TOKEN` does not exist in your Colab secrets.
To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.
You will be able to reuse this secret in all of your notebooks.
Please note that authentication is recommended but still optional to access public models or datasets.
  warnings.warn(


config.json:   0%|          | 0.00/632 [00:00<?, ?B/s]

pytorch_model.bin.index.json:   0%|          | 0.00/26.8k [00:00<?, ?B/s]

pytorch_model-00001-of-00002.bin:   0%|          | 0.00/9.98G [00:00<?, ?B/s]

pytorch_model-00002-of-00002.bin:   0%|          | 0.00/3.50G [00:00<?, ?B/s]

Loading checkpoint shards:   0%|          | 0/2 [00:00<?, ?it/s]

generation_config.json:   0%|          | 0.00/174 [00:00<?, ?B/s]

/usr/local/lib/python3.10/dist-packages/transformers/generation/configuration_utils.py:563: UserWarning: `do_sample` is set to `False`. However, `temperature` is set to `0.9` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `temperature`. This was detected when initializing the generation config instance, which means the corresponding file may hold incorrect parameterization and should be fixed.
  warnings.warn(
/usr/local/lib/python3.10/dist-packages/transformers/generation/configuration_utils.py:568: UserWarning: `do_sample` is set to `False`. However, `top_p` is set to `0.6` -- this flag is only used in sample-based generation modes. You should set `do_sample=True` or unset `top_p`. This was detected when initializing the generation config instance, which means the corresponding file may hold incorrect parameterization and should be fixed.
  warnings.warn(
/usr/local/lib/python3.10/dist-packages/transformers/generation/configuratio

tokenizer_config.json:   0%|          | 0.00/695 [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/1.84M [00:00<?, ?B/s]

special_tokens_map.json:   0%|          | 0.00/434 [00:00<?, ?B/s]

# Using Fine-tuned Model for Prediction

In [6]:
from transformers import pipeline

print('---------------------------------------------------------------------------------')
print('-----Welcome to the ChemProtBot. You are now ready to start asking questions?----')
print('---------------------------------------------------------------------------------')

pipe = pipeline(task="text-generation", model=model,
                tokenizer=tokenizer, max_length=500,
                do_sample=True, temperature=0.9,
                top_p=0.6, truncation=True)

while True:
    query = input(f"\nPrompt: ")

    if query.lower() in {"exit", "quit", "q", "f"}:
        print('Exiting')
        break

    if query.strip() == '':
        continue

    result = pipe(f"<s>[INST] {query} [/INST]")
    print(f"\nAnswer: " + result[0]['generated_text'].split('[/INST]  ')[1])

---------------------------------------------------------------------------------
-----Welcome to the ChemProtBot. You are now ready to start asking questions?----
---------------------------------------------------------------------------------

Prompt: Which proteins are mentioned in the following text? pmid:26467040. Hypolipidemic drugs, such as bezafibrate, known activators of the peroxisome proliferator-activated receptor α (PPARα), induce both the pantothenate kinase 1 (PANK1) gene and miR107 (∼2.5-fold) in primary human hepatocytes. Surprisingly, CYP2C8 mRNA and protein levels were induced by bezafibrate, known activators of the peroxisome proliferator-activated receptor α (PPARα),. CYP2C8 promoter activity was increased by ectopic expression of PPARα in HepG2 cells, with a further increase after bezafibrate (∼18-fold), 4-chloro-6-(2,3-xylidino)-2-pyrimidinylthio acetic acid (∼10-fold) treatment, or the antidiabetic drug rosiglitazone, all known CYP2C8 activators. Chromatin immu

Truncation was not explicitly activated but `max_length` is provided a specific value, please use `truncation=True` to explicitly truncate examples to max length. Defaulting to 'longest_first' truncation strategy. If you encode pairs of sequences (GLUE-style) with the tokenizer you can select this strategy more precisely by providing a specific strategy to `truncation`.
/usr/local/lib/python3.10/dist-packages/transformers/generation/utils.py:1347: UserWarning: You have modified the pretrained model configuration to control generation. This is a deprecated strategy to control generation and will be removed soon, in a future version. Please use and modify the model generation configuration (see https://huggingface.co/docs/transformers/generation_strategies#default-text-generation-configuration )
  warnings.warn(
Starting from v4.46, the `logits` model output will have the same type as the model (except at train time, where it will always be FP32)



Answer: Based on the provided text, the following proteins are mentioned:

1. PPARα (peroxisome proliferator-activated receptor alpha)
2. PANK1 (pantothenate kinase 1)
3. miR107 (microRNA 107)
4. CYP2C8 (cytochrome P450 2C8)
5. PPARα response element (PRASE)

Prompt: Which chemicals are mentioned in the following text? pmid:26467040. Hypolipidemic drugs, such as bezafibrate, known activators of the peroxisome proliferator-activated receptor α (PPARα), induce both the pantothenate kinase 1 (PANK1) gene and miR107 (∼2.5-fold) in primary human hepatocytes. Surprisingly, CYP2C8 mRNA and protein levels were induced by bezafibrate, known activators of the peroxisome proliferator-activated receptor α (PPARα),. CYP2C8 promoter activity was increased by ectopic expression of PPARα in HepG2 cells, with a further increase after bezafibrate (∼18-fold), 4-chloro-6-(2,3-xylidino)-2-pyrimidinylthio acetic acid (∼10-fold) treatment, or the antidiabetic drug rosiglitazone, all known CYP2C8 activators.

##### Which proteins are mentioned in the following text? pmid:26467040. Hypolipidemic drugs, such as bezafibrate, known activators of the peroxisome proliferator-activated receptor α (PPARα), induce both the pantothenate kinase 1 (PANK1) gene and miR107 (∼2.5-fold) in primary human hepatocytes. Surprisingly, CYP2C8 mRNA and protein levels were induced by bezafibrate, known activators of the peroxisome proliferator-activated receptor α (PPARα),. CYP2C8 promoter activity was increased by ectopic expression of PPARα in HepG2 cells, with a further increase after bezafibrate (∼18-fold), 4-chloro-6-(2,3-xylidino)-2-pyrimidinylthio acetic acid (∼10-fold) treatment, or the antidiabetic drug rosiglitazone, all known CYP2C8 activators. Chromatin immunopreciptation assay analysis confirmed recruitment of PPARα to a PPARα response element located at position -2109 base pair relative to the translation start site of CYP2C8 after bezafibrate treatment of human hepatocytes.

##### Which chemicals are mentioned in the following text? pmid:26467040. Hypolipidemic drugs, such as bezafibrate, known activators of the peroxisome proliferator-activated receptor α (PPARα), induce both the pantothenate kinase 1 (PANK1) gene and miR107 (∼2.5-fold) in primary human hepatocytes. Surprisingly, CYP2C8 mRNA and protein levels were induced by bezafibrate, known activators of the peroxisome proliferator-activated receptor α (PPARα),. CYP2C8 promoter activity was increased by ectopic expression of PPARα in HepG2 cells, with a further increase after bezafibrate (∼18-fold), 4-chloro-6-(2,3-xylidino)-2-pyrimidinylthio acetic acid (∼10-fold) treatment, or the antidiabetic drug rosiglitazone, all known CYP2C8 activators. Chromatin immunopreciptation assay analysis confirmed recruitment of PPARα to a PPARα response element located at position -2109 base pair relative to the translation start site of CYP2C8 after bezafibrate treatment of human hepatocytes.

##### What is the relationship between Bezafibrate and PPARα in the following text? pmid:26467040. Hypolipidemic drugs, such as bezafibrate, known activators of the peroxisome proliferator-activated receptor α (PPARα), induce both the pantothenate kinase 1 (PANK1) gene and miR107 (∼2.5-fold) in primary human hepatocytes. Surprisingly, CYP2C8 mRNA and protein levels were induced by bezafibrate, known activators of the peroxisome proliferator-activated receptor α (PPARα),. CYP2C8 promoter activity was increased by ectopic expression of PPARα in HepG2 cells, with a further increase after bezafibrate (∼18-fold), 4-chloro-6-(2,3-xylidino)-2-pyrimidinylthio acetic acid (∼10-fold) treatment, or the antidiabetic drug rosiglitazone, all known CYP2C8 activators. Chromatin immunopreciptation assay analysis confirmed recruitment of PPARα to a PPARα response element located at position -2109 base pair relative to the translation start site of CYP2C8 after bezafibrate treatment of human hepatocytes.